lib/task_helpers/exports/customization_templates.rb

```diff
@@ -0,0 +1,33 @@
+module TaskHelpers
+  class Exports
+    class CustomizationTemplates
+      EXCLUDE_ATTRS = %i(created_at updated_at id pxe_image_type_id class).freeze
+      def export(options = {})
+        export_dir = options[:directory]
+
+        customization_templates = options[:all] ? CustomizationTemplate.all : CustomizationTemplate.where(:system => [false, nil])
+
+        customization_templates.order(:id).each do |customization_template|
+          $log.info("Exporting Customization Template: #{customization_template.name} (ID: #{customization_template.id})")
+
+          ct_hash = Exports.exclude_attributes(customization_template.to_model_hash, EXCLUDE_ATTRS)
+          ct_hash.merge!(pxe_image(customization_template.pxe_image_type))
+
+          image_type_name = ct_hash[:pxe_image_type][:name] ? ct_hash[:pxe_image_type][:name] : "Examples"
+          filename = Exports.safe_filename("#{image_type_name}-#{ct_hash[:name]}", options[:keep_spaces])
+          File.write("#{export_dir}/#{filename}.yaml", ct_hash.to_yaml)
+        end
+      end
+
+      private
+
+      def pxe_image(pxe_image)
+        if pxe_image
+          { :pxe_image_type => pxe_image.to_model_hash.reject { |key| EXCLUDE_ATTRS.include?(key) } }
+        else
+          { :pxe_image_type => {} }
```

Comment:
Is this something you ran into?  Looks like it is an invalid configuration so we are just exporting something we know will fail to import.

Is there a reason to export or should the code detect and skip these records during export?

Reply:
This is to get around the issue that `.to_model_hash` outputs a key called `:pxe_image_type_id` and we need to convert that ID into the `:name` and `provision_type` from the PxeImageType that the ID is for so that the import can work between regions. The import script looks for a key called `:pxe_image_type` and coverts it into a valid PxeImageType object that is then used when finding the CustomiztionTemplate to update or creating a new CustomizationTemplate object.

Having the empty hash is there to support exporting the examples. The examples don't have a `:pxe_image_type_id`, but for the import to work I need to have the `:pxe_image_type` key so I just set it to an empty hash. The import script aborts with an error if the `:pxe_image_type` does not contain a key named `:name`.